Signal/src/view controllers/MessageComposeTableViewController.m

```diff
@@ -407,17 +487,34 @@ - (void)messageComposeViewController:(MFMessageComposeViewController *)controlle
 #pragma mark - Table View Data Source
 
 - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
-    return 2;
+    return 4;
 }
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
```

Comment:
This method controls which one (if any) of the "special" controls are shown.  Hopefully its readable.